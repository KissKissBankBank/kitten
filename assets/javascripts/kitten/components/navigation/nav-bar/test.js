import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { NavBar } from './index'

describe('<NavBar />', () => {
  let component

  describe('simple NavBar', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <NavBar
            className="customClass"
            listProps={{ className: 'customListClass' }}
            navProps={{ className: 'customNavClass' }}
          >
            <NavBar.ListItem linkProps={{ href: '#1' }} active>
              Item 1
            </NavBar.ListItem>
            <NavBar.ListItem
              className="customItemClass"
              linkProps={{ href: '#2', className: 'customLinkClass' }}
            >
              Item 2
            </NavBar.ListItem>
            <NavBar.ListItem linkProps={{ href: '#3' }}>Item 3</NavBar.ListItem>
            <NavBar.ListItem linkProps={{ href: '#4' }}>Item 4</NavBar.ListItem>
          </NavBar>,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('NavBar with `colors` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <NavBar
            className="customClass"
            listProps={{ className: 'customListClass' }}
            navProps={{ className: 'customNavClass' }}
            colors={{
              background: '#fff',
              link: '#00f',
              border: 'transparent',
              activeLink: '#808',
              activeBorder: 'rgba(128,0,128,.5)',
            }}
          >
            <NavBar.ListItem linkProps={{ href: '#1' }} active>
              Item 1
            </NavBar.ListItem>
            <NavBar.ListItem
              className="customItemClass"
              linkProps={{ href: '#2', className: 'customLinkClass' }}
            >
              Item 2
            </NavBar.ListItem>
            <NavBar.ListItem linkProps={{ href: '#3' }}>Item 3</NavBar.ListItem>
            <NavBar.ListItem linkProps={{ href: '#4' }}>Item 4</NavBar.ListItem>
          </NavBar>,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
