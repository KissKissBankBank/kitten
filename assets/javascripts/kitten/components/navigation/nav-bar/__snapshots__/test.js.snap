// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> NavBar with \`colors\` prop matches with snapshot 1`] = `
.c0 {
  width: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 .k-NavBar__nav {
  width: auto;
  margin: 0;
  background: var(--navBar-background);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0.k-NavBar--hasPadding .k-NavBar__nav {
  padding: 0 1.25rem;
}

.c0.k-NavBar--hasPadding .k-NavBar__nav::before,
.c0.k-NavBar--hasPadding .k-NavBar__nav::after {
  content: '';
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 .k-NavBar__list {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.5625rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  padding: 0;
}

.c0 .k-NavBar__link {
  height: 4.0625rem;
  box-sizing: border-box;
  font-family: Maax,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  color: var(--navBar-link);
  border-bottom: 0.25rem solid var(--navBar-border);
  -webkit-transition: color 0.2s,border-color 0.2s;
  transition: color 0.2s,border-color 0.2s;
  cursor: pointer;
}

.c0 .k-NavBar__link:hover {
  color: var(--navBar-activeLink);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .k-NavBar__link:focus,
.c0 .k-NavBar__link[aria-current='page'] {
  border-color: var(--navBar-activeBorder);
  color: var(--navBar-activeLink);
  border-width: 0.25rem;
}

.c0.k-NavBar--small .k-NavBar__list {
  gap: 1.875rem;
}

.c0.k-NavBar--small .k-NavBar__link {
  height: 3.75rem;
}

.c0.k-NavBar--big .k-NavBar__link,
.c0.k-NavBar--large .k-NavBar__link {
  height: 5rem;
}

.c0.k-NavBar--center .k-NavBar__nav {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0.k-NavBar--start .k-NavBar__nav {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0.k-NavBar--end .k-NavBar__nav {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0.k-NavBar--space-between .k-NavBar__nav {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:640px) {
  .c0.k-NavBar--hasPadding .k-NavBar__nav {
    padding: 0 2.5rem;
  }
}

@media (min-width:640px) {
  .c0 .k-NavBar__list {
    gap: 3.125rem;
  }
}

@media (min-width:640px) {
  .c0 .k-NavBar__link {
    height: 5rem;
  }
}

@media (min-width:640px) {
  .c0.k-NavBar--big .k-NavBar__link,
  .c0.k-NavBar--large .k-NavBar__link {
    height: 6.25rem;
  }
}

@media (max-width:639px) {
  .c0.k-NavBar--mobile-center .k-NavBar__nav {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c0.k-NavBar--mobile-start .k-NavBar__nav {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }

  .c0.k-NavBar--mobile-end .k-NavBar__nav {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }

  .c0.k-NavBar--mobile-space-between .k-NavBar__nav {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<div
  className="c0 k-NavBar k-NavBar--medium k-NavBar--center customClass k-NavBar--hasPadding"
  style={
    Object {
      "--navBar-activeBorder": "rgba(128,0,128,.5)",
      "--navBar-activeLink": "#808",
      "--navBar-background": "#fff",
      "--navBar-border": "transparent",
      "--navBar-link": "#00f",
    }
  }
>
  <nav
    className="k-NavBar__nav customNavClass"
  >
    <ul
      className="k-NavBar__list customListClass"
    >
      <li
        className="k-NavBar__listItem"
      >
        <a
          aria-current="page"
          className="k-NavBar__link"
          href="#1"
        >
          Item 1
        </a>
      </li>
      <li
        className="k-NavBar__listItem customItemClass"
      >
        <a
          aria-current={null}
          className="k-NavBar__link customLinkClass"
          href="#2"
        >
          Item 2
        </a>
      </li>
      <li
        className="k-NavBar__listItem"
      >
        <a
          aria-current={null}
          className="k-NavBar__link"
          href="#3"
        >
          Item 3
        </a>
      </li>
      <li
        className="k-NavBar__listItem"
      >
        <a
          aria-current={null}
          className="k-NavBar__link"
          href="#4"
        >
          Item 4
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<NavBar /> NavBar with \`modifier\` prop matches with snapshot 1`] = `
.c0 {
  width: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 .k-NavBar__nav {
  width: auto;
  margin: 0;
  background: var(--navBar-background);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0.k-NavBar--hasPadding .k-NavBar__nav {
  padding: 0 1.25rem;
}

.c0.k-NavBar--hasPadding .k-NavBar__nav::before,
.c0.k-NavBar--hasPadding .k-NavBar__nav::after {
  content: '';
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 .k-NavBar__list {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.5625rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  padding: 0;
}

.c0 .k-NavBar__link {
  height: 4.0625rem;
  box-sizing: border-box;
  font-family: Maax,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  color: var(--navBar-link);
  border-bottom: 0.25rem solid var(--navBar-border);
  -webkit-transition: color 0.2s,border-color 0.2s;
  transition: color 0.2s,border-color 0.2s;
  cursor: pointer;
}

.c0 .k-NavBar__link:hover {
  color: var(--navBar-activeLink);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .k-NavBar__link:focus,
.c0 .k-NavBar__link[aria-current='page'] {
  border-color: var(--navBar-activeBorder);
  color: var(--navBar-activeLink);
  border-width: 0.25rem;
}

.c0.k-NavBar--small .k-NavBar__list {
  gap: 1.875rem;
}

.c0.k-NavBar--small .k-NavBar__link {
  height: 3.75rem;
}

.c0.k-NavBar--big .k-NavBar__link,
.c0.k-NavBar--large .k-NavBar__link {
  height: 5rem;
}

.c0.k-NavBar--center .k-NavBar__nav {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0.k-NavBar--start .k-NavBar__nav {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0.k-NavBar--end .k-NavBar__nav {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0.k-NavBar--space-between .k-NavBar__nav {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:640px) {
  .c0.k-NavBar--hasPadding .k-NavBar__nav {
    padding: 0 2.5rem;
  }
}

@media (min-width:640px) {
  .c0 .k-NavBar__list {
    gap: 3.125rem;
  }
}

@media (min-width:640px) {
  .c0 .k-NavBar__link {
    height: 5rem;
  }
}

@media (min-width:640px) {
  .c0.k-NavBar--big .k-NavBar__link,
  .c0.k-NavBar--large .k-NavBar__link {
    height: 6.25rem;
  }
}

@media (max-width:639px) {
  .c0.k-NavBar--mobile-center .k-NavBar__nav {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c0.k-NavBar--mobile-start .k-NavBar__nav {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }

  .c0.k-NavBar--mobile-end .k-NavBar__nav {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }

  .c0.k-NavBar--mobile-space-between .k-NavBar__nav {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<div
  className="c0 k-NavBar k-NavBar--large k-NavBar--center customClass k-NavBar--hasPadding"
  style={
    Object {
      "--navBar-activeBorder": "var(--color-primary-500)",
      "--navBar-activeLink": "var(--color-primary-500)",
      "--navBar-background": "var(--color-grey-000)",
      "--navBar-border": "transparent",
      "--navBar-link": "var(--color-grey-900)",
    }
  }
>
  <nav
    className="k-NavBar__nav customNavClass"
  >
    <ul
      className="k-NavBar__list customListClass"
    >
      <li
        className="k-NavBar__listItem"
      >
        <a
          aria-current="page"
          className="k-NavBar__link"
          href="#1"
        >
          Item 1
        </a>
      </li>
      <li
        className="k-NavBar__listItem customItemClass"
      >
        <a
          aria-current={null}
          className="k-NavBar__link customLinkClass"
          href="#2"
        >
          Item 2
        </a>
      </li>
      <li
        className="k-NavBar__listItem"
      >
        <a
          aria-current={null}
          className="k-NavBar__link"
          href="#3"
        >
          Item 3
        </a>
      </li>
      <li
        className="k-NavBar__listItem"
      >
        <a
          aria-current={null}
          className="k-NavBar__link"
          href="#4"
        >
          Item 4
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<NavBar /> simple NavBar matches with snapshot 1`] = `
.c0 {
  width: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 .k-NavBar__nav {
  width: auto;
  margin: 0;
  background: var(--navBar-background);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0.k-NavBar--hasPadding .k-NavBar__nav {
  padding: 0 1.25rem;
}

.c0.k-NavBar--hasPadding .k-NavBar__nav::before,
.c0.k-NavBar--hasPadding .k-NavBar__nav::after {
  content: '';
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 .k-NavBar__list {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.5625rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  padding: 0;
}

.c0 .k-NavBar__link {
  height: 4.0625rem;
  box-sizing: border-box;
  font-family: Maax,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  color: var(--navBar-link);
  border-bottom: 0.25rem solid var(--navBar-border);
  -webkit-transition: color 0.2s,border-color 0.2s;
  transition: color 0.2s,border-color 0.2s;
  cursor: pointer;
}

.c0 .k-NavBar__link:hover {
  color: var(--navBar-activeLink);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .k-NavBar__link:focus,
.c0 .k-NavBar__link[aria-current='page'] {
  border-color: var(--navBar-activeBorder);
  color: var(--navBar-activeLink);
  border-width: 0.25rem;
}

.c0.k-NavBar--small .k-NavBar__list {
  gap: 1.875rem;
}

.c0.k-NavBar--small .k-NavBar__link {
  height: 3.75rem;
}

.c0.k-NavBar--big .k-NavBar__link,
.c0.k-NavBar--large .k-NavBar__link {
  height: 5rem;
}

.c0.k-NavBar--center .k-NavBar__nav {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0.k-NavBar--start .k-NavBar__nav {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0.k-NavBar--end .k-NavBar__nav {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0.k-NavBar--space-between .k-NavBar__nav {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:640px) {
  .c0.k-NavBar--hasPadding .k-NavBar__nav {
    padding: 0 2.5rem;
  }
}

@media (min-width:640px) {
  .c0 .k-NavBar__list {
    gap: 3.125rem;
  }
}

@media (min-width:640px) {
  .c0 .k-NavBar__link {
    height: 5rem;
  }
}

@media (min-width:640px) {
  .c0.k-NavBar--big .k-NavBar__link,
  .c0.k-NavBar--large .k-NavBar__link {
    height: 6.25rem;
  }
}

@media (max-width:639px) {
  .c0.k-NavBar--mobile-center .k-NavBar__nav {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c0.k-NavBar--mobile-start .k-NavBar__nav {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }

  .c0.k-NavBar--mobile-end .k-NavBar__nav {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }

  .c0.k-NavBar--mobile-space-between .k-NavBar__nav {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<div
  className="c0 k-NavBar k-NavBar--medium k-NavBar--center customClass k-NavBar--hasPadding"
  style={
    Object {
      "--navBar-activeBorder": "var(--color-primary-500)",
      "--navBar-activeLink": "var(--color-primary-500)",
      "--navBar-background": "var(--color-grey-000)",
      "--navBar-border": "transparent",
      "--navBar-link": "var(--color-grey-900)",
    }
  }
>
  <nav
    className="k-NavBar__nav customNavClass"
  >
    <ul
      className="k-NavBar__list customListClass"
    >
      <li
        className="k-NavBar__listItem"
      >
        <a
          aria-current="page"
          className="k-NavBar__link"
          href="#1"
        >
          Item 1
        </a>
      </li>
      <li
        className="k-NavBar__listItem customItemClass"
      >
        <a
          aria-current={null}
          className="k-NavBar__link customLinkClass"
          href="#2"
        >
          Item 2
        </a>
      </li>
      <li
        className="k-NavBar__listItem"
      >
        <a
          aria-current={null}
          className="k-NavBar__link"
          href="#3"
        >
          Item 3
        </a>
      </li>
      <li
        className="k-NavBar__listItem"
      >
        <a
          aria-current={null}
          className="k-NavBar__link"
          href="#4"
        >
          Item 4
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
